src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -103,32 +136,42 @@ private void AppendValue(ViewBufferValue value)
         private ViewBufferPage GetCurrentPage()
         {
             ViewBufferPage page;
-            if (Pages.Count == 0)
```

Comment:
Would it be easier to share the code between this piece and `MoveTo` if we separated the addition of pages and keeping track of where to write next:

```C#
public ViewBufferPage GetCurrentPage()
{
    if (_currentPage == null || _currentPage.IsFull)
    {
        _currentPage = new ViewBufferPage(_bufferScope.GetPage(_pageSize));
        AddPage(_currentPage);
    }
    
    return _currentPage;
}

public void AddPage(ViewBufferPage page)
{
    if (_multiplePages != null)
    {
        _multiplePages.Add(page);
    }
    else if (_singlePage != null)
    {
        _multiplePages = new List<ViewBufferPage>(2);
        _multiplePages.Add(_singlePage);
        _multiplePages.Add(page);
    }
    else
    {
        _singlePage = page;
    }
}

```